### DISCUSSION TITLE
Forms\Components\Select displaying ID for selected record on modifyQueryUsing

### DISCUSSION THREAD

**Discussion by yuvrajtimalsina** [filamentphp/filament#8359] [General]
### Select dropdown on modifying query displays ID for selected record. 

![screenshot](https://github.com/filamentphp/filament/assets/107990171/b777a609-a35f-4c3f-9d6e-5abb6d6d44e5)

**Here's my code :** 

```
  Forms\Components\Select::make('banner_id')
                            ->placeholder('Select Banner')
                            ->preload()
                            ->searchable()
                            ->relationship(
                                'banner',
                                'title',
                                modifyQueryUsing: fn (EloquentBuilder $query) => $query->whereDoesntHave('page'))
                            ->native(false),
```

**Comment by leandrocfe** [comment]
Maybe this?
```php
->relationship(
    name: 'banner',
    titleAttribute: 'title',
    modifyQueryUsing: fn (Builder $query) => $query->whereDoesntHave('page'),
)
```
Does the select work if you remove modifyQueryUsing?

**Reply by yuvrajtimalsina** [reply]
It works in both cases as the data being saved is an ID.The thing is its displaying ID for selected record while on edit page instead of 'title'.

**Reply by leandrocfe** [reply]
Console error?

**Reply by yuvrajtimalsina** [reply]
No console error as well since its working, just need to display title instead of ID there.

**Reply by leandrocfe** [reply]
This was supposed to work. Can you share the project on github?

**Reply by yuvrajtimalsina** [reply]
I asked my collegue to replicate the issue and have provided you access in your email - leandrocfe@gmail.com
 https://github.com/neerozprasain/guru-demo

**Reply by leandrocfe** [reply]
You have a logic issue. You are displaying banners in the select field that do not have a corresponding page, even though you already have a page saved. For example, your select field has '1' as the saved value, but your options are:

`2 => 'Option 2',
3 => 'Option 3',`

If you want to show available options, you should do

```php
->relationship(
    'banner',
    'title',
    modifyQueryUsing: function (Builder $query, ?Page $record) {
        $query
            ->when($record, fn (Builder $query) => $query->whereId($record->banner_id))
            ->orWhere(function (Builder $query) {
                $query->whereDoesntHave('page');
            });
    }
)
```

**Reply by yuvrajtimalsina** [reply]
Thank you for your time and response, it means alot. I was supposed to display banners that weren't already associated with any page.I will implement the logic as you suggested later on and ping here with an update accordingly.
